### PR TITLE
Changes needed by java-binding for now.

### PR DESCRIPTION
**Pull request by beeender**
This PR is not ready to be merged to the master since we haven't decided how to deal with async-query and others.
This PR exposes more things to java-binding to use `SharedRealm` and current java async query implementation.
Create a PR to make changes more visible since object-store is a submodule in realm-java, those changes are not visible to realm-java PR.

### CODE REVIEW

**Comment by kneth** [comment]
Looks good to me

File: src/impl/collection_change_builder.cpp [hunk 1]
```diff
@@ -142,7 +142,12 @@ void CollectionChangeBuilder::parse_complete()
     for (auto move : m_move_mapping) {
         REALM_ASSERT_DEBUG(deletions.contains(move.second));
         REALM_ASSERT_DEBUG(insertions.contains(move.first));
-        moves.push_back({move.second, move.first});
+        if (move.first == move.second) {
+            deletions.remove(move.second);
+            insertions.remove(move.first);
+        }
+        else
```
Comment (kneth) [inline src/impl/collection_change_builder.cpp]: Please use `{}`

Comment (beeender) [inline src/impl/collection_change_builder.cpp]: Uh... it is merged from #158 to solve the build problems with 2.0.0-rc4 ... We are not using those code core :) @kneth

**Comment by beeender** [comment]
using `realm-java` branch instead.

**Pull request closed without merging**